incubator/hnc/internal/validators/object_test.go

```diff
@@ -493,6 +500,59 @@ func TestUserChanges(t *testing.T) {
 	}
 }
 
+// fakeClient implements a fake client (https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#Client).
+type fakeClient struct {
```

Comment:
Make `fakeNSClient` since (as mentioned above) it has to actually return a namespace object?